code/controllers/subsystem/input.dm

```diff
@@ -34,6 +52,55 @@ SUBSYSTEM_DEF(input)
 		var/client/user = clients[i]
 		user.set_macros()
 
+///queue a click from usr onto clicked_atom with the given mouse handling arguments. only works if usr is the player mob clicking.
+/datum/controller/subsystem/input/proc/queue_click_from_usr(atom/clicked_atom, atom/location, control, params)
+	if(control != "mapwindow.map" || !ismob(usr) || QDELING(usr) || QDELETED(clicked_atom))
+		return FALSE
+
+	//high priority because clicks should be as low latency as possible, deferring to the beginning of the next SSinput run should rarely delay
+	//a click by more than a few milliseconds, but since the MC almost always resumes after every other sleeping proc this isnt guaranteed
```

Comment:
How is it possible for SSinput to delay a click? it should be fired immediately as the next tick starts shouldn't it?

Reply:
not always because the MC loop isnt the only sleeping proc that exists, and most other sleeping procs will run before Master in the tick theyre scheduled to start because Master almost always sleeps for 1 tick and other procs usually sleep for longer which puts them further in the queue.

its totally possible for a ton of continually sleeping, expensive procs to delay the MC resuming a relevant amount. if that happens we stop queuing clicks until the delay averages down below the threshold again (though i should make the average MC_AVG_FAST_UP_SLOW_DOWN())

Reply:
Ah I see, the old meteor behavior